drivers/soundwire/cadence_master.c

```diff
@@ -2449,6 +2449,43 @@ int sdw_cdns_prepare_read_dma_buffer(u8 dev_num, u32 start_register, int data_si
 		if (ret < 0)
 			return ret;
 
+		counter++;
+
+		p_dma_buffer += dma_data_written;
+		dma_buffer_size -= dma_data_written;
+		total_dma_data_written += dma_data_written;
+	}
+
+	/* Add fack frame */
+	header[0] &= ~GENMASK(7, 6);	/* Mark inactive to the fack frame */
```

Comment:
you should explain what a 'fake frame' is, looks like you are using a flag in the BPT header.